stage1/init/common/pod.go

```diff
@@ -794,3 +794,22 @@ func protectKernelTunables(opts []*unit.UnitOption, appName types.ACName, system
 
 	return opts
 }
+
+// generateWritablePaths appends the list of options required to mark
+// a rootfs RO and specific paths as RW, according to runtime options.
+func generateWritablePaths(opts []*unit.UnitOption, podAbsRoot string, runApp *schema.RuntimeApp, vols map[types.ACName]types.Volume, mounts []mountWrapper) ([]*unit.UnitOption, error) {
```

Comment:
Actually thinking more, I don't see the necessity of this function at all, since we already have `appendOptionsList` which could be easily modified to take a `vals` vararg.